discovery/gossiper_test.go

```diff
@@ -105,40 +105,48 @@ func (n *mockSigner) SignMessage(pubKey *btcec.PublicKey,
 }
 
 type mockGraphSource struct {
-	nodes      []*channeldb.LightningNode
-	infos      map[uint64]*channeldb.ChannelEdgeInfo
-	edges      map[uint64][]*channeldb.ChannelEdgePolicy
 	bestHeight uint32
+
+	mu    sync.Mutex
+	nodes []channeldb.LightningNode
```

Comment:
probably safer to keep this as `[]*channeldb.LightningNode`, otherwise we will be returning pointers to the raw backing array. Mutations on the objects will be written through to the slice, which breaks the guarantees of the locking. with a pointer, the actual object can be modified without introducing race conditions. same comment goes for the edge policy slice

Reply:
The pointer needs to be removed, otherwise, due to the asynchronous nature of the gossiper, it will try to read/write the same value. This is only a problem with the mock graph implementation - in real world scenarios this doesn't happen due to the DB returning pointers for new objects. I still see what you mean though, so I've modified the mock implementation to return the pointer of a copy.